components/servo_tracing/src/lib.rs

```diff
@@ -0,0 +1,319 @@
+                        span,
+                        "Attribute argument `name` has been provided more than once",
+                    ));
+                },
+                arg @ _ => {
+                    eprintln!("{:?}", arg.path().to_token_stream().to_string());
+                    passthrough_args.push(arg);
+                },
+            }
+        }
+
+        Ok(InstrumentConfiguration {
+            passthrough_args,
+            field_args,
+            level,
+            name,
+        })
+    }
+}
+
+/// injects servo_profiling as needed
+fn inject_servo_profiling(items: MetaList) -> syn::Result<MetaList> {
+    let metas: Punctuated<Meta, Comma> = Punctuated::parse_terminated.parse2(items.tokens)?;
+
+    let mut servo_profiling_given = false;
+    let mut empty = true;
+
+    for arg in metas.iter() {
+        servo_profiling_given |= arg.path().is_ident("servo_profiling");
+        empty = false;
+    }
+
+    let items = match (servo_profiling_given, empty) {
+        (true, _) => parse_quote! { fields(#metas) },
+        (false, true) => parse_quote! {
+            fields(servo_profiling = true)
+        },
+        (false, false) => {
+            parse_quote! {
+                fields(#metas, servo_profiling = true)
+            }
+        },
+    };
+
+    Ok(items)
+}
+
+fn instrument_internal(
+    attr: proc_macro2::TokenStream,
+    item: proc_macro2::TokenStream,
+) -> syn::Result<proc_macro2::TokenStream> {
+    dbg!(&attr);
+    // Prepare passthrough arguments for tracing::instrument
+    let InstrumentConfiguration {
+        mut passthrough_args,
+        field_args,
+        level,
+        name,
+    } = parse2(attr)?;
+    let input_fn: ItemFn = parse2(item)?;
+
+    // setup invariants to make reconstruction easier
+    let field_args = field_args.unwrap_or_else(|| parse_quote! { fields(servo_profiling = true) });
+    let field_args = inject_servo_profiling(field_args)?;
+    // invariant, field_args is `fields(...)` and `servo_profiling = true` is injected unless explicitly
+    // set otherwise
+    let level = level.unwrap_or(parse2(quote! {"trace"}).unwrap());
+    passthrough_args.push(parse2(quote! {level = #level}).unwrap());
+    // invariant, passthrough args has >= 1 items
+    let passthrough_args = quote! {
+        #(#passthrough_args),*,
+    };
+    // invariant, passthrough_args has a comma
+    let name = match name {
+        Some(name) => quote! { #name, },
+        None => quote! {},
+    };
+    // invariant, name has a comma if needed
+
+    // reconstruct
+    let fn_attrs: &Vec<Attribute> = &input_fn.attrs;
+    let fn_vis = &input_fn.vis;
+    let fn_sig = &input_fn.sig;
+    let fn_block = &input_fn.block;
+
+    let output = quote! {
+        #[cfg_attr(
+            feature = "tracing",
+            tracing::instrument(
+                #name
+                #passthrough_args
+                #field_args
+            )
+        )]
+        #(#fn_attrs)*
```

Comment:
Can't we just put the original `item` here and skip all the reconstructing? We don't want to change anything about the original function (including the existing other attributes), right?